lib/geocoder/lookups/google_places_search.rb

```diff
@@ -18,16 +18,42 @@ def supported_protocols
 
       private
 
+      def results(query)
+        super(query, 'candidates')
+      end
+
       def base_query_url(query)
-        "#{protocol}://maps.googleapis.com/maps/api/place/textsearch/json?"
+        "#{protocol}://maps.googleapis.com/maps/api/place/findplacefromtext/json?"
       end
 
       def query_url_google_params(query)
         {
-          query: query.text,
+          input: query.text,
+          inputtype: 'textquery',
+          fields: fields(query),
           language: query.language || configuration.language
         }
       end
+
+      def fields(query)
```

Comment:
By default every fields are requested, to prevent any regression from developers already using the gem.